pkg/app/context.go

```diff
@@ -328,18 +328,28 @@ func (ctx *RequestContext) GetIndex() int8 {
 	return ctx.index
 }
 
+func (ctx *RequestContext) SetIndex(index int8) {
+	ctx.index = index
+}
+
 type HandlerFunc func(c context.Context, ctx *RequestContext)
 
 // HandlersChain defines a HandlerFunc array.
 type HandlersChain []HandlerFunc
 
 var handlerNames = make(map[uintptr]string)
 
+var handlerRWLock sync.RWMutex
+
 func SetHandlerName(handler HandlerFunc, name string) {
+	handlerRWLock.Lock()
```

Comment:
Is there a bench test comparing with sync.Map? Probably paste some bench test stats in this PR?

Reply:
+1

Reply:
goos: darwin
goarch: amd64
pkg: github.com/cloudwego/hertz/pkg/app
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
BenchmarkInbuiltHandlerNameOperator-12           5482314               201.1 ns/op
BenchmarkConcurrentHandlerNameOperator-12        5449671               222.2 ns/op
PASS
ok      github.com/cloudwego/hertz/pkg/app      5.130s